src/Sound/Tidal/UI.hs

```diff
@@ -1439,6 +1445,50 @@ _arp name p = arpWith f p
         thumbup xs = concatMap (\x -> [thumb,x]) $ tail xs
           where thumb = head xs
 
+{- | `rolled` plays each note of a chord quickly in order, as opposed to simultaneously; to give a chord a harp-like effect.
+This will played from the lowest note to the highest note of the chord
+@
+rolled $ n "c'maj'4" # s "superpiano"
+@
+
+You can reverse this order by using the `rolled'` function
+
+@
+rolled' $ n "c'maj9" # s "superpiano"
+@
+
+And you can use `rolledBy` or `rolledBy'` to specify the length of the roll. The value in the passed pattern
+is the divisor of the cycle length.
+
+@
+rolledBy "<1 0.5 0.25 0.125>" $ note "c'maj9" # s "superpiano"
+@
+-}
+
+rolledWith :: ([Event a] -> [EventF (ArcF Time) a]) -> Ratio Integer -> Pattern a -> Pattern a
+rolledWith f t = withEvents aux
+         where aux es = concatMap (steppityIn) (groupBy (\a b -> whole a == whole b) $ (f es))
+               steppityIn xs = mapMaybe (\(n, ev) -> (timeguard n xs ev t)) $ enumerate xs
```

Comment:
I added this zero check to make this possible: 

```
s "superpiano" <| n (rolledBy "0" "[0,1,2,3]")
```

Otherwise we would receive an exception anyways.